src/rtp/ext.rs

```diff
@@ -635,8 +710,95 @@ pub struct ExtensionValues {
     pub mid: Option<Mid>,
     #[doc(hidden)]
     pub frame_mark: Option<u32>,
+
+    /// User values for [`ExtensionSerializer`] to parse into and write from.
+    pub user_values: UserExtensionValues,
+}
+
+// The "AnyMap" idea is borrowed from the http crate but replacing Box for Any.
+type AnyMap = HashMap<TypeId, Arc<dyn Any + Send + Sync>, BuildHasherDefault<IdHasher>>;
+
+// No point in hashing the TypeId, since it is already unique.
+#[derive(Default)]
+struct IdHasher(u64);
+
+impl Hasher for IdHasher {
+    fn write(&mut self, _: &[u8]) {
+        unreachable!("TypeId calls write_u64");
+    }
+
+    #[inline]
+    fn write_u64(&mut self, id: u64) {
+        self.0 = id;
+    }
+
+    #[inline]
+    fn finish(&self) -> u64 {
+        self.0
+    }
 }
 
+/// Space for storing user extension values via [`ExtensionSerializer`].
+#[derive(Clone, Default)]
+pub struct UserExtensionValues {
+    map: Option<AnyMap>,
+}
+
+// TODO: I don't see a good way of comparing this. Is there one?
+impl PartialEq for UserExtensionValues {
```

Comment:
When do we compare `ExtensionValues`?

We can compare by comparing keys + the objects pointed to by the `Arc`s

Reply:
I tried to remove PartialEq/Eq from `ExtensionValues`. Since they are in `RtpHeader` they are also in `RtpPacket` and thus used quite extensively in our unit tests.

Remove them would break public API.

We can compare keys, but the objects pointed to by the Arc seems kinda pointless.

Reply:
I guess all that matters is that you have registered the same serializers so sure let's skip comparing the pointers

Reply:
We can check that both sides of the eq has a certain TypeId. Beyond that I think we can only compare the pointer addresses, not the values pointed to.

This seems to be impossible: `v1.downcast_ref::<PartialEq>()` because PartialEq takes an argument.

Reply:
Alright. Let's check type id.